### PR TITLE
Update to latest dependency versions

### PR DESCRIPTION
**Pull request by redonkulus**
@lkraav @ericf 

This updates almost all of the dependencies. Express 4.x would be done in a new PR since it requires more changes, just getting to latest 3.x for now. `es6-module-transpiler` is also out of date but no longer maintained. Path to upgrade is `babel` but that would be another PR as well.

### CODE REVIEW

**Comment by yahoocla** [comment]
CLA is valid!

File: tests/unit/middleware.error.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ describe('Error cases', function () {
         it('should respond with 404', function (done) {
             request(app).get('/404')
                 .expect(404)
-                .expect('Content-Type', 'text/html; charset=UTF-8')
+                .expect('Content-Type', 'text/html; charset=utf-8')
```
Comment (redonkulus) [inline tests/unit/middleware.error.js]: these are mainly just changes with latest express

**Comment by ericf** [comment]
The Node 4 build in Travis seems to be failing, while the others pass. @redonkulus do you want to either remove Node 4 from Travis and just have Node 6 LTS, or dig into what's going on?

**Comment by redonkulus** [comment]
@ericf the issue I believe is that grunt-broccoli-build is not compatible with grunt@1.x. Fixing that packages peerDeps would resolve the issue. I believe you own this package 😀

**Comment by redonkulus** [comment]
@ericf installing npm@3 in travis fixes the peerDep issue. This is good to review now.

**Comment by ericf** [comment]
👍 @redonkulus do you want to squash with the green button?

**Pull request merged**